### PR TITLE
Make the nuget packages cache directory repo local for local and CI B…

### LINKED ISSUE #80: Make CI Build NuGet Cache Independent
See https://github.com/dotnet/cli/issues/3368 for an explanation of the issues present.

### PR DESCRIPTION
**Pull request by brthor**
This makes nuget local to the repo directory in CI and local builds.

Fixes #80 

cc @eerhardt @livarcocc

### CODE REVIEW

**Comment by eerhardt** [comment]
:shipit:

**Comment by livarcocc** [comment]
CI is failing to resolve packages:

/mnt/resource/j/workspace/dotnet_core-setup/release_1.0.0/debug_opensuse13.2_x64_prtest/tools/independent/RuntimeGraphGenerator/project.json(10,62): error NU1001: The dependency System.Xml.XmlSerializer >= 4.0.11-rc3-24201-00 could not be resolved.

File: build_projects/shared-build-targets-utils/Utils/Dirs.cs [hunk 1]
```diff
@@ -47,9 +47,9 @@ private static string GetNuGetPackagesDir()
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                return Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".nuget", "packages");
+                return Path.Combine(Dirs.RepoRoot, ".nuget", "packages");
```
Comment (livarcocc) [inline build_projects/shared-build-targets-utils/Utils/Dirs.cs]: Check this: https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs#L159.

You don't necessarily need to use Dirs.RepoRoot. You can look at the env variable here as well.

**Comment by brthor** [comment]
bad ordering of the targets, fix is up

File: build_projects/shared-build-targets-utils/Utils/Dirs.cs (hunk 1 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Dirs.cs]: Not sure what you mean, Dirs.RepoRoot is what makes the cache inside the repo so we don't have build-to-build conflicts.

The method you linked will read from the environment variable that we set here.

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by eerhardt** [comment]
:shipit: 

what was the OSX failure?

**Comment by brthor** [comment]
offline machine

**Pull request merged**